src/power_grid_model/utils.py

```diff
@@ -398,3 +404,61 @@ def self_test():
             print("Self test finished.")
         except Exception as e:
             raise PowerGridError from e
+
+
+def _make_test_case(
+    save_path: Path,
+    input_data: SingleDataset,
+    params: dict,
```

Comment:
As a follow up, we may want to create a proper `class`or `enum` or something more specific rather than a `dict`for the paramenters. If the goal is to make users' life easier, and ours, this would be a good next step.

Reply:
what about `**kwargs`? e.g. 
```suggestion
    calculation_type: CalculationType,
    *,
    rtol: float = 1e-8,
    atol: float = 1e-6,
    experimental_features: _ExperimentalFeatures | None = None,
    extra_params: dict = dict(),
```

Reply:
> what about `**kwargs`? e.g.

The overload of params makes it less ideal to go kwargs way for the `_make_test_case` function of this PR and for the upcoming `validate_test_case` function. I don't think there is a good way of implementing this overload logic.

We can either choose 
* keep params as dict and not have a special support for it.
* Change structure of the params to be a collection of multiple calculation params based on calculation method.